### PR TITLE
PLANET-4803: Convert Cookies Block to WYSIWYG

### PR DESCRIPTION
**Pull request by pablocubico**
Ref: https://jira.greenpeace.org/browse/PLANET-4803

A few notes on this:
My first prototype was using `react-cookie` but I found out that doubled our bundle size for `frontendIndex.js`, going from around 40kb to around 90kb...

As I was porting the code I realized this has a counterpart on the Master theme, where the "cookie notice" UI lives. There are also some utility functions there that I needed for the blocks side, so I replicated those methods from the master theme (readCookie, createCookie...), .

Instead of duplicating that code, maybe we could do something as waiting for those global methods to be available using a Timeout, which is a widely used pattern (or anti-pattern...), see: https://gist.github.com/chrisjhoughton/7890239 . But I think that's uglier than actually replicating these methods in the blocks code.

Note that I pinned the `main` script from master theme as a dependency for `frontendIndex` but that's not enough, because even when it does alter the order of script inclusion from WP, it does not ensure that the methods will be available before executing the second one...

### CODE REVIEW

**Comment by lithrel** [comment]
Ah sorry, forgot this: it is hard to actually click on _necessary cookie name_ and _all cookies name_ in the editor. It works from read mode to edit mode, but once you're in edit mode you can only get in those fields by using `tab`. I had this problem on another PR, due to a [very wide rule](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/314/files#diff-ae6da3ef197a199dc3b2672693070c19R33) on pointer-events; I think the same problem applies here (pointer-events: none), although not because of the same origin, but my attempts to solve this ended up in errors. Maybe it should be solved in another ticket.

File: assets/src/blocks/Cookies/CookiesFrontend.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
+import { Component, Fragment } from '@wordpress/element';
+import { FrontendRichText } from '../../components/FrontendRichText/FrontendRichText';
+
+const { __ } = wp.i18n;
+
+export class CookiesFrontend extends Component {
+  constructor(props) {
+		super(props);
```
[the remark this answers is not in the document]
Comment (mleray) [reply assets/src/blocks/Cookies/CookiesFrontend.js]: Should be good now! 😄

**Comment by mleray** [comment]
> Ah sorry, forgot this: it is hard to actually click on _necessary cookie name_ and _all cookies name_ in the editor. It works from read mode to edit mode, but once you're in edit mode you can only get in those fields by using `tab`. I had this problem on another PR, due to a [very wide rule](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/314/files#diff-ae6da3ef197a199dc3b2672693070c19R33) on pointer-events; I think the same problem applies here (pointer-events: none), although not because of the same origin, but my attempts to solve this ended up in errors. Maybe it should be solved in another ticket.

I'm not sure I understand what you mean by this...? I don't have any issues accessing those fields in the editor 🤔

**Comment by lithrel** [comment]
> I'm not sure I understand what you mean by this...? I don't have any issues accessing those fields in the editor 🤔

As we discussed, this is a problem occuring on firefox and safari, works properly on chrome. I will create a ticket for this and check if some react solutions exist.

**Pull request merged**